File: solutions/javascript/bob/1/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  const trimmedMessage = message.trim();

  if (trimmedMessage === '') {
    return "Fine. Be that way!";
  }
  
  const isQuestion = trimmedMessage.endsWith('?');
  
  const hasLetters = /[A-Za-z]/.test(trimmedMessage);
  const isYelling = hasLetters && trimmedMessage === trimmedMessage.toUpperCase();
  
  if (isYelling && isQuestion) {
    return "Calm down, I know what I'm doing!";
  }
  
  if (isYelling) {
    return "Whoa, chill out!";
  }
  
  if (isQuestion) {
    return "Sure.";
  }
  
  return "Whatever.";
};

File: solutions/python/bob/1/bob.py
def response(hey_bob):
    def is_question(msg):
        return msg.strip().endswith("?")

    def is_yell(msg):
        msg = msg.strip()
        return msg.isupper() and any(c.isalpha() for c in msg)

    def is_silence(msg):
        return msg.strip() == ""

    hey_bob = hey_bob.strip()

    if is_question(hey_bob) and is_yell(hey_bob):
        return "Calm down, I know what I'm doing!" 
    elif is_question(hey_bob):
        return "Sure."
    elif is_yell(hey_bob):
        return "Whoa, chill out!"
    elif is_silence(hey_bob):
        return "Fine. Be that way!"

    return "Whatever."